dev/src/path.ts

```diff
@@ -477,20 +481,29 @@ export class FieldPath extends Path<FieldPath> {
    */
   static validateFieldPath(fieldPath: string|FieldPath) {
     if (!(fieldPath instanceof FieldPath)) {
-      if (!is.string(fieldPath)) {
+      if (fieldPath === undefined) {
```

Comment:
FWIW this seems to contradict the type signature of the method, which is weird / confusing.  Perhaps we should be handling undefined higher up, or we should fix the signature to indicate that we don't actually know what fieldPath is?